### ISSUE TITLE
A puzzle about the 2.4.1 and 2.6.1 getDependencyList API change

### ISSUE THREAD

**Issue by justinplus** [os72/protobuf-dynamic#13]
Hi,

There is a comment about the ProtoBuf API change in `DynamicSchema.java`.

```java
// getDependencyList() signature was changed and broke compatibility in 2.6.1; workaround with reflection
//List<String> dependencyList = fdProto.getDependencyList();
```
link: https://github.com/os72/protobuf-dynamic/blob/master/src/main/java/com/github/os72/protobuf/dynamic/DynamicSchema.java#L223-L224

I just checked this method signature in ProtoBuf 2.4.1 and 2.6.1

**In 2.4.1**
```java
public java.util.List<String> getDependencyList() {
      return dependency_;
}
```
**in 2.6.1**
```java
public com.google.protobuf.ProtocolStringList getDependencyList() {
      return dependency_;
}
```
but `ProtocolStringList` is actually a `List<String>`
```java
public interface ProtocolStringList extends List<String> {

  /** Returns a view of the data as a list of ByteStrings. */
  List<ByteString> asByteStringList();

}
```

So what does the broken compatibility here mean?

Thank you,
Justin

**Comment by os72** [comment]
For some reason it was not possible to compile with 2.4.1 and run against 2.6.1. I agree with your observation and I was puzzled too. Try it out

**Comment by justinplus** [comment]
Changing the return type does break the binary compatibility as showed in this stack overflow question.
https://stackoverflow.com/questions/47476509/can-i-change-a-return-type-to-be-a-strict-subtype-and-retain-binary-compatibilit

And I noticed that it's a fix when your repo still compiled with 2.4.1.
> 228e0fb: workaround 2.6.1 compatibility issue: use reflection to call fdProto.getDependencyList()

Maybe better to just use direct method call since your repo has upgraded to 2.6.1.

**Comment by os72** [comment]
Yes good point, since the older 2.4.1-compatible version is now on a separate branch

**Comment by os72** [comment]
Released 1.0.1, 0.9.5
